### ISSUE TITLE
Nested properties of non-strict schema are undefined after reload

### ISSUE THREAD

**Issue by adamreisnz** [Automattic/mongoose#11309]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
Nested properties of a non-strict schema appear as undefined after a document is loaded from the database

**If the current behavior is a bug, please provide the steps to reproduce.**
```
const mongoose = require('mongoose')
const {Schema} = mongoose
const uri = `mongodb://localhost:27017/test`

mongoose.set('debug', true)
console.log(`Mongoose version: ${mongoose.version}`)

//Schema
const NestedSchema = new Schema({}, {
  id: false,
  _id: false,
  strict: false,
})

const ItemSchema = new Schema({
  name: {
    type: String,
  },
  nested: NestedSchema,
})

//Model
const Item = mongoose.model('Item', ItemSchema)

//Run function
async function run() {
  await mongoose.connect(uri)

  //Create new item
  const item = await Item.create({
    name: 'Test',
    nested: {
      something: 1,
      else: 'test',
      foo: {
        bar: 55,
      },
    },
  })

  console.log(`After creation: ${item.nested.foo.bar}`)

  //Modify nested data
  item.nested.foo.bar = 66
  item.markModified('nested.foo.bar')
  await item.save()

  console.log(`After save: ${item.nested.foo.bar}`)

  //Reload
  const reloaded = await Item.findOne({_id: item._id})
  console.log(`After reload: ${reloaded.nested}`)
  console.log(`After reload: ${reloaded.nested.foo}`)
  console.log(`After reload: ${reloaded.nested.foo.bar}`)
}

//Run
run().catch(err => console.log(err))
```

**What is the expected behavior?**
To be able to log the nested property `foo` and see it's value after reloading the item from the database.
Instead, `foo` is undefined.

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**
Mongoose 6.1.10
MongoDB 5.0.3
Node 16.13.1

**Comment by adamreisnz** [comment]
Thanks @vkarpov15, so this was a confirmed issue?

**Comment by vkarpov15** [comment]
@adamreisnz yes this is a confirmed bug and will be fixed in v6.2.1
